File: src/BjjEire.Domain/Entities/Common/PricingModel.cs
using BjjEire.Domain.Enums;

namespace BjjEire.Domain.Entities.Common;

public class PricingModel {
    [BsonElement("type")]
    public PricingType Type { get; set; }

    [BsonElement("amount")]
    public decimal Amount { get; set; }

    [BsonElement("durationDays")]
    public int? DurationDays { get; set; }

    [BsonElement("urrency")]
    public string? Currency { get; set; } = "EUR";
}

File: src/bjjeire-api/src/main/java/com/bjjeire/api/event/PricingModel.java
package com.bjjeire.api.event;

import java.math.BigDecimal;
import java.util.List;

public record PricingModel(
        PricingType type,
        String label,
        List<BjjEventType> appliesToTypes,
        BigDecimal amount,
        Integer durationDays,
        String currency) {}
